atomic_reactor/tasks/binary.py

```diff
@@ -0,0 +1,103 @@
+"""
+Copyright (c) 2021 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+
+from dataclasses import dataclass
+from typing import ClassVar
+
+from atomic_reactor import source
+from atomic_reactor.tasks import common
+from atomic_reactor.tasks import plugin_based
+
+
+@dataclass(frozen=True)
+class BinaryBuildTaskParams(common.TaskParams):
+    """Binary build task parameters (this task only uses common parameters)."""
+
+    # Validate with the source containers schema instead
+    user_params_schema: ClassVar[str] = "schemas/user_params.json"
+
+    @property
+    def source(self) -> source.Source:
+        """Return a dummy source for a binary container build."""
+        return source.DummySource(None, None, workdir=self.build_dir)
+
+
+class PreBinaryBuildTask(plugin_based.PluginBasedTask):
+    """Binary container pre-build task."""
+
+    plugins_def = plugin_based.PluginsDef(
+        prebuild=[
+            {"name": "check_user_settings"},
+            {"name": "check_and_set_platforms"},
+            {"name": "flatpak_create_dockerfile"},
+            {"name": "add_flatpak_labels"},
+            {"name": "inject_parent_image"},
+            {"name": "pull_base_image"},
+            {"name": "koji_parent"},
+            {"name": "resolve_composes"},
+            {"name": "add_image_content_manifest"},
+            {"name": "add_filesystem"},
+            {"name": "flatpak_update_dockerfile"},
+            {"name": "bump_release"},
+            {"name": "add_labels_in_dockerfile"},
+            {"name": "resolve_remote_source"},
+            {"name": "pin_operator_digest"},
+            {"name": "hide_files"},
+            {"name": "distribution_scope"},
+            {"name": "add_buildargs_in_dockerfile"},
+            {"name": "pull_base_image"},
+            {"name": "change_from_in_dockerfile"},
+            {"name": "add_help"},
+            {"name": "fetch_maven_artifacts"},
+            {"name": "add_dockerfile"},
+            {"name": "distgit_fetch_artefacts"},
+            {"name": "inject_yum_repos"},
+        ],
```

Comment:
@tkdchen could you double-check that the plugins were split into tasks in the right order? I think there is at least one problem, `fetch_maven_artifacts` and `resolve_remote_source` both need to run before `add_image_content_manifest`

Reply:
@hjmodi I updated the plugin list of this Pre task. There is a slight change to the order.

Reply:
updated the order here as well